bloomberg/bcpc
# Upgrade Zabbix to 2.4

A motivation for upgrading Zabbix now is the new [ability to remove missing resources](https://www.zabbix.com/documentation/2.4/manual/introduction/whatsnew240#option_to_remove_missing_resources_when_importing) when importing Zabbix configuration. As monitoring checks are expected to continually evolve, this provides additional convenience. #728 is an example. For more comprehensive list of changes, see https://www.zabbix.com/documentation/2.4/manual/introduction/whatsnew240.

This also requires upgrading pyzabbix and actually pull it from upstream, instead of copying into our repo which I mistakenly thought it was our code when I reinstated it from our old branches.

This should be tested with #744 as otherwise, errors (though harmless on multi-node cluster) may be thrown during API requests.

Upgrade procedure on existing monitoring cluster:
* Stop Zabbix server and Apache on all monitoring nodes
<pre>
stop zabbix-server
service apache2 stop
</pre>
* Run chef on each node. On the first node, it will upgrade the zabbix packages _and_ database schema. I began the upgrade on the monitoring VIP holder.

Some verification steps to run:
1. Review https://10.0.100.6/zabbix/tr_status.php for errors. The list should only list Zabbix agent version changes and a Ceph health warning.
2. Check agent and server logs in `/var/log/zabbix` for errors.
3. Fail over Zabbix server instance to other monitoring nodes to ensure they are just as functional. Do this by stopping keepalived on the VIP holder.
4. Check that each Zabbix frontend (http://10.0.100.14:7777, http://10.0.100.15:7777, http://10.0.100.16:7777) loads the login page.
